### PR TITLE
Armory

### PR DESCRIPTION
**Pull request by bhollis**
Still a bit rough, but this represents a lot of work already. Clicking the item header now shows this sheet instead of opening DestinyTracker. It has links to other sites, and a bunch of DIM info including every wishlist roll (which is super unweildy...).

Obvious stuff for the future is adding perk editing w/ stats preview, and wishlist editing (plus some better way of showing wishlists) but I'm just getting started.

Screen Shot 2021-09-13 at 6 51 26 PM
Screen Shot 2021-09-13 at 6 51 31 PM
Screen Shot 2021-09-13 at 6 51 44 PM

### CODE REVIEW

**Comment by inexorableAce** [comment]
small issue on windows chrome. Otherwise this looks awesome. 
image

**Review by inexorableAce — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -71,13 +67,15 @@
     "dequips",
     "dim",
     "dim's",
+    "Dont",
```
Comment (48klocs) [inline .vscode/settings.json:70]: Is this correct?

File: .vscode/settings.json [hunk 2]
```diff
@@ -104,11 +102,14 @@
...
     "rahool",
     "reddit",
+    "Reforgeable",
     "shaxx",
     "stackable",
     "teespring",
+    "Unacquired",
     "unequip",
     "untrack",
+    "Upgradei",
```
Comment (48klocs) [inline .vscode/settings.json:112]: This also seems like a typo.

**Review by 48klocs — APPROVED** [review]
Nice! I think this is going to be a pretty killer feature for folks sharing screenshots of their shiny toys.

File: .vscode/settings.json (hunk 1 above)
Comment (delphiactual) [reply .vscode/settings.json:70]: DontForgetDupes

File: .vscode/settings.json (hunk 2 above)
Comment (delphiactual) [reply .vscode/settings.json:112]: UpgradeiOS

File: .vscode/settings.json (hunk 1 above)
Comment (48klocs) [reply .vscode/settings.json:70]: I figured there was a good reason for them. Thanks!

**Review by delphiactual — APPROVED** [review]

File: src/app/armory/Links.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      `https://data.destinysets.com/i/InventoryItem:${item.hash}?lang=${language}`,
+    hideOnPhone: true,
+  },
+];
+
+export default function Links({ item }: { item: DimItem }) {
+  const language = useSelector(languageSelector);
+  const isPhonePortrait = useIsPhonePortrait();
+  return (
+    <ul className={styles.links}>
```
Comment (nev-r) [inline src/app/armory/Links.tsx:41]: the inconsistent underlined space here is a bit weird
image

File: src/app/armory/Links.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  }
+
+  return `https://destinytracker.com/destiny-2/db/items/${item.hash}${perkQueryString}`;
+}
+
+/**
+ * Banshee-44 puts placeholder entries in for the still-mysterious socketTypeHash 0.
+ * If you look at Scathelocke https://data.destinysets.com/i/InventoryItem:3762467078
+ * for one example, socketEntires[5] has a socketTypeHash of 0. We discard this
```
Comment (nev-r) [inline src/app/armory/Links.tsx:79]: *socketEntries

a fun fact: this typehash 0 on all weapons, is the now nulled-out element socket that you could use to swap a weapon's element in Y1

File: src/app/armory/Links.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    if (perkCsv?.length) {
+      perkQueryString = `?perks=${perkCsv}`;
+    }
+  }
+
+  return `https://destinytracker.com/destiny-2/db/items/${item.hash}${perkQueryString}`;
+}
+
+/**
+ * Banshee-44 puts placeholder entries in for the still-mysterious socketTypeHash 0.
```
Comment (nev-r) [inline src/app/armory/Links.tsx:77]: this confused me until i realized you're talking about the website? or maybe you aren't? do you mean the vendor-offered fixed rolls? guess i am still confused.

**Review by nev-r — APPROVED** [review]
feel weird about it not being sheet shaped but it's whatever. cool place to do a lot of new stuff.

File: src/app/armory/Links.tsx (hunk 5 above)
Comment (bhollis) [reply src/app/armory/Links.tsx:77]: Someone else wrote this originally but I suspect it was from the banshee-44 site that let you mess with rolls.

File: src/app/armory/Links.tsx (hunk 3 above)
Comment (bhollis) [reply src/app/armory/Links.tsx:41]: Yeah I wish links were a bit more subtle about underlining

File: src/app/armory/Links.tsx (hunk 5 above)
Comment (48klocs) [reply src/app/armory/Links.tsx:77]: That was me - Banshee-44 let you configure weapons + rolls, and generated a URL to share that particular combination with other folks, and I used that so I didn't feel bad about not building out way more tooling to let people pick their own wish list items.

So file it under legacy support.

**Pull request merged**